doc/py_tutorials/py_setup/py_setup_in_ubuntu/py_setup_in_ubuntu.markdown

```diff
@@ -0,0 +1,242 @@
+
+*(All the below commands can be done in a single cmake statement, but it is split here for better
+understanding.)*
+
+-   Enable TBB and Eigen support:
+    <br>
+    `cmake -D WITH_TBB=ON -D WITH_EIGEN=ON ..`
+    <br>
+-   Enable documentation and disable tests and samples<br>
+    
+    `cmake -D BUILD_DOCS=ON -D BUILD_TESTS=OFF -D BUILD_PERF_TESTS=OFF -D BUILD_EXAMPLES=OFF ..`<br>
+   <br>
+-   Disable all GPU related modules.<br>
+    <br>
+    `cmake -D WITH_OPENCL=OFF -D WITH_CUDA=OFF -D BUILD_opencv_gpu=OFF -D BUILD_opencv_gpuarithm=OFF -D BUILD_opencv_gpubgsegm=OFF -D BUILD_opencv_gpucodec=OFF -D BUILD_opencv_gpufeatures2d=OFF -D BUILD_opencv_gpufilters=OFF -D BUILD_opencv_gpuimgproc=OFF -D BUILD_opencv_gpulegacy=OFF -D BUILD_opencv_gpuoptflow=OFF -D BUILD_opencv_gpustereo=OFF -D BUILD_opencv_gpuwarping=OFF ..`
+    <br>
+-   Set installation path and build type<br>
+    <br>
+    `cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local ..`
+   <br>
+Each time you enter cmake statement, it prints out the resulting configuration setup. In the final
+setup you got, make sure that following fields are filled (below is the some important parts of
+configuration I got). These fields should be filled appropriately in your system also. Otherwise
+some problem has happened. So check if you have correctly performed above steps.
+@code{.sh}
+...
+--   GUI:
+--     GTK+ 2.x:                    YES (ver 2.24.19)
+--     GThread :                    YES (ver 2.36.3)
+
+--   Video I/O:
+--     DC1394 2.x:                  YES (ver 2.2.0)
+--     FFMPEG:                      YES
+--       codec:                     YES (ver 54.92.100)
+--       format:                    YES (ver 54.63.104)
+--       util:                      YES (ver 52.18.100)
+--       swscale:                   YES (ver 2.2.100)
+--       gentoo-style:              YES
+--     GStreamer:
+--       base:                      YES (ver 0.10.36)
+--       video:                     YES (ver 0.10.36)
+--       app:                       YES (ver 0.10.36)
+--       riff:                      YES (ver 0.10.36)
+--       pbutils:                   YES (ver 0.10.36)
+
+--     V4L/V4L2:                    Using libv4l (ver 1.0.0)
+
+--   Other third-party libraries:
+--     Use Eigen:                   YES (ver 3.1.4)
+--     Use TBB:                     YES (ver 4.0 interface 6004)
+
+--   Python:
+--     Interpreter:                 /usr/bin/python2 (ver 2.7.5)
+--     Libraries:                   /lib/libpython2.7.so (ver 2.7.5)
+--     numpy:                       /usr/lib/python2.7/site-packages/numpy/core/include (ver 1.7.1)
+--     packages path:               lib/python2.7/site-packages
+
+...
+@endcode
+Many other flags and settings are there. It is left for you for further exploration.
+<br>
+Now you build the files using make command and install it using make install command. make install
+should be executed as root.<br>
+@code{.sh}
+make
+su
+make install
```

Comment:
`sudo make install`

But this way is no technically correct. Use `checkinstall` to prepare Ubuntu packages and install them separately.